pkg/util/promtest/histogram.go

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+package promtest
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/prometheus/client_golang/prometheus"
+	dto "github.com/prometheus/client_model/go"
+)
+
+// IsNativeHistogram checks if a gathered metric has positive or negative spans.
+//
+// This is possible because native histograms without any observations and zero_threshold of 0
+// use a no-op span (offset 0, length 0).
+func IsNativeHistogram(g prometheus.Gatherer, metricName string) error {
+	family, err := gatherMetricFamily(g, metricName)
+	if err != nil {
+		return err
+	}
+	histograms, err := getHistograms(family)
+	if err != nil {
+		return err
+	}
+	for i, h := range histograms {
+		if len(h.PositiveSpan)+len(h.NegativeSpan) == 0 {
```

Comment:
I think  https://github.com/prometheus/client_golang/blob/b157309b723f0b8588ea604bb78dbbba196803f2/prometheus/histogram.go#L846 means you need to also check for `&& h.ZeroThreshold==0 && h.ZeroCount==0` and the float version as well if it has one.

Because that referenced code allows for the zero bucket to be used, but no spans. https://prometheus.io/docs/specs/native_histograms/#zero-bucket